packages/stark-ui/src/modules/transform-input-directive/directives/transform-input.directive.ts

```diff
@@ -0,0 +1,154 @@
+
+/**
+ * Provider for StarkTransformInputDirective
+ */
+export const STARK_TRANSFORM_INPUT_PROVIDER: any = {
+	provide: NG_VALUE_ACCESSOR,
+	// tslint:disable-next-line:no-forward-ref
+	useExisting: forwardRef(() => StarkTransformInputDirective),
+	multi: true
+};
+// tslint:disable:jsdoc-format
+/**
+ * Directive to transform the input value of an input or textarea. It sits between the native element and the ngControl (ngModel/formControl) so observers are triggered only once.
+ * This directive will not affect changes applied trough code, only user input will be transformed. This means i.a. initial values are not transformed.
+ * 
+ * It is possible to pass 'uppercase', 'lowercase' or a custom function to transform the value. This directive works if the input is controlled or not.
+ * **examples:**
+  ```html
+   <input [(ngModel)] StarkTransformInputDirective="uppercase" />
+   // OR
+   <input [formControl] StarkTransformInputDirective="lowercase" />
+   // OR
+   <input StarkTransformInputDirective="someFunction" />
+  ```
+ */
+// tslint:enable:jsdoc-format
+@Directive({
+	// tslint:disable-next-line:directive-selector
+	selector: "[starkTransformInput]",
+	providers: [STARK_TRANSFORM_INPUT_PROVIDER],
+	host: {
+		"(input)": "_onInput($event)",
+		"(blur)": "_onTouched()"
+	}
+})
+export class StarkTransformInputDirective implements ControlValueAccessor, OnChanges {
+	/**
+	 * Internal property for holding the transformation function
+	 */
+	public _transformation: (value: any) => any;
+
+	// tslint:disable-next-line:no-input-rename
+	@Input("starkTransformInput")
+	public set transformation(transformation: StarkInputTransformationType) {
+		switch (transformation) {
+			case UPPERCASE:
+				this._transformation = (v: string) => v.toUpperCase();
+				break;
+			case LOWERCASE:
+				this._transformation = (v: string) => v.toLocaleLowerCase();
+				break;
+			default:
+				this._transformation = transformation;
+				break;
+		}
+	}
+
+	/**
+	 * The registered callback function called when an input event occurs on the input element.
+	 */
+	private _onChange: (_: any) => void = (_: any) => {
+		/*noop*/
+	};
+
+	/**
+	 * @ignore
+	 * The registered callback function called when a blur event occurs on the input element.
+	 */
+	public _onTouched: () => void = () => {
+		/*noop*/
+	};
+
+	/**
+	 * Class constructor
+	 * @param _renderer - Angular renderer
+	 * @param _elementRef - Reference to the element
+	 */
+	public constructor(private _renderer: Renderer2, private _elementRef: ElementRef) {}
+
+	/**
+	 * Angular life cycle hook
+	 * @param changes - simple changes
+	 */
+	public ngOnChanges(changes: SimpleChanges): void {
+		const currentValue: any = changes.transformation.currentValue;
+		// Type guard
+		if (typeof currentValue !== "function" && ![UPPERCASE, LOWERCASE].includes(currentValue)) {
+			throw Error('directive starkTransformInput should be of type "uppercase" | "lowercase" | ((value: any) => any)');
```

Comment:
I suggest to change this error to: "[starkTransformInput]: the transformation input is not valid. It should be of type StarkInputTransformationType"

So this message will remain valid even if we augment/adapt the `StarkInputTransformationType`